webapi/Controllers/ChatMemoryController.cs

```diff
@@ -52,19 +53,24 @@ public class ChatMemoryController : ControllerBase
     /// <param name="chatId">The chat id.</param>
     /// <param name="memoryName">Name of the memory type.</param>
     [HttpGet]
-    [Route("chatMemory/{chatId:guid}/{memoryName}")]
+    [Route("chatMemory/{chatId:guid}/{memoryType}")]
     [ProducesResponseType(StatusCodes.Status200OK)]
     [ProducesResponseType(StatusCodes.Status400BadRequest)]
     [Authorize(Policy = AuthPolicyName.RequireChatParticipant)]
     public async Task<IActionResult> GetSemanticMemoriesAsync(
         [FromServices] ISemanticMemoryClient memoryClient,
         [FromRoute] string chatId,
-        [FromRoute] string memoryName)
+        [FromRoute] string memoryType)
     {
         // Sanitize the log input by removing new line characters.
         // https://github.com/microsoft/chat-copilot/security/code-scanning/1
         var sanitizedChatId = GetSanitizedParameter(chatId);
-        var sanitizedMemoryName = GetSanitizedParameter(memoryName);
+
+        if (!this._promptOptions.TryGetMemoryName(memoryType, out string memoryName))
```

Comment:
What's the difference between memoryType and memoryName?

Reply:
MemoryType maps to an enum, and the values of that are stable. Memory name is configurable in appsettings.json and maps to the name of the index or tag used in the memory store. o the idea is that the API contract uses memoryType because it won't have it's values changed, and then the memoryName can be changed in appsettings.json without breaking anything.